projects/nginx/build.sh

```diff
@@ -0,0 +1,142 @@
+cd /nginx
+
+# Apply diff for libFuzzer and AFL++ builds
+hg import /add_fuzzers.diff --no-commit
+cp /make_fuzzers auto/make_fuzzers
+
+cd src/fuzz
+rm -rf genfiles && mkdir genfiles && /LPM/external.protobuf/bin/protoc http_request_proto.proto --cpp_out=genfiles
+cd ../..
+
+# Build targets for libfuzzer
+export CC=clang
+export CXX=clang++
+export CFLAGS="-g -fsanitize=fuzzer-no-link,address,integer,bounds,null,undefined,float-divide-by-zero"
+export CXXFLAGS="-g -fsanitize=fuzzer-no-link,address,integer,bounds,null,undefined,float-divide-by-zero"
+
+auto/configure \
+    --with-http_v2_module
+make -f objs/Makefile modules
+
+export CC=clang
+export CXX=clang++
+export CFLAGS="-g -fsanitize=fuzzer,address,integer,bounds,null,undefined,float-divide-by-zero"
+export CXXFLAGS="-g -fsanitize=fuzzer,address,integer,bounds,null,undefined,float-divide-by-zero"
+make -f objs/Makefile fuzzers
+
+for fuzzer in objs/*_fuzzer;
+do
+    fuzzer_name=$(basename $fuzzer)
+    fuzzer_name=${fuzzer_name::-7}
+    cp $fuzzer /${fuzzer_name}_lf
+done
+
+# Build targets for AFL++
+export CC=afl-clang-fast
+export CXX=afl-clang-fast++
+export CFLAGS="-g -fsanitize=address,integer,bounds,null,undefined,float-divide-by-zero"
+export CXXFLAGS="-g -fsanitize=address,integer,bounds,null,undefined,float-divide-by-zero"
+
+auto/configure \
+    --with-http_v2_module
+make -f objs/Makefile modules
+
+export CC=afl-clang-fast
+export CXX=afl-clang-fast++
+export CFLAGS="-g -fsanitize=fuzzer,address,integer,bounds,null,undefined,float-divide-by-zero"
+export CXXFLAGS="-g -fsanitize=fuzzer,address,integer,bounds,null,undefined,float-divide-by-zero"
+
+make -f objs/Makefile fuzzers
+
+for fuzzer in objs/*_fuzzer;
+do
+    fuzzer_name=$(basename $fuzzer)
+    fuzzer_name=${fuzzer_name::-7}
+    cp $fuzzer /${fuzzer_name}_afl
+done
+
+# Apply diff for Sydr and cov builds
+cd ../nginx_sydr
+hg import /add_sydr.diff --no-commit
+cp /make_sydr auto/make_sydr
+
+cd src/fuzz
+rm -rf genfiles && mkdir genfiles && /LPM/external.protobuf/bin/protoc http_request_proto.proto --cpp_out=genfiles
+cd ../..
+
+# Build targets for Sydr
+export CC=clang
+export CXX=clang++
+export CFLAGS="-g -fPIE"
+export CXXFLAGS="-g -fPIE"
+
+auto/configure \
+    --with-http_v2_module
+
+make -f objs/Makefile fuzzers
+
+for fuzzer in objs/*_fuzzer;
+do
+    fuzzer_name=$(basename $fuzzer)
+    fuzzer_name=${fuzzer_name::-7}
+    cp $fuzzer /${fuzzer_name}_sydr
+done
+
+# Build targets for cov
+export CC=clang
+export CXX=clang++
+export CFLAGS="-g -fprofile-instr-generate -fcoverage-mapping -fPIE"
+export CXXFLAGS="-g -fprofile-instr-generate -fcoverage-mapping -fPIE"
+
+auto/configure \
+    --with-http_v2_module
+
+make -f objs/Makefile fuzzers
+
+for fuzzer in objs/*_fuzzer;
+do
+    fuzzer_name=$(basename $fuzzer)
+    fuzzer_name=${fuzzer_name::-7}
+    cp $fuzzer /${fuzzer_name}_cov
+done
+
+# Prepare seed corpora
+cd /
+git clone https://github.com/dvyukov/go-fuzz-corpus.git /go-fuzz-corpus
```

Comment:
You can delete `go-fuzz-corpus` directory after corpora preparation. git clone can be also moved to Dockerfile as it is done in other oss-sydr-fuzz projects.

Reply:
Done